test/test_integration_cluster.rb

```diff
@@ -21,11 +21,33 @@ def teardown
   end
 
   def test_hot_restart_does_not_drop_connections_threads
-    hot_restart_does_not_drop_connections num_threads: 10, total_requests: 3_000
+    restart_does_not_drop_connections num_threads: 10, total_requests: 3_000,
+      signal: :USR2
   end
 
   def test_hot_restart_does_not_drop_connections
-    hot_restart_does_not_drop_connections num_threads: 1, total_requests: 1_000
+    restart_does_not_drop_connections num_threads: 1, total_requests: 1_000,
+      signal: :USR2
+  end
+
+  def test_phased_restart_does_not_drop_connections_threads
+    restart_does_not_drop_connections num_threads: 10, total_requests: 3_000,
+      signal: :USR1
+  end
+
+  def test_phased_restart_does_not_drop_connections
+    restart_does_not_drop_connections num_threads: 1, total_requests: 1_000,
+      signal: :USR1
+  end
+
+  def test_phased_restart_does_not_drop_connections_threads_fork_worker
+    restart_does_not_drop_connections num_threads: 10, total_requests: 3_000,
+      signal: :USR1 #, config: 'fork_worker', log: true
```

Comment:
Was this intentionally commented out? It seems like fork worker should be explicitly enabled here.

Reply:
The test seems to time out when enabling `fork_worker` here, see https://github.com/puma/puma/pull/3297#issuecomment-2509608503.

I think we should bump [this sleep](https://github.com/puma/puma/pull/3557/files#diff-c117a0d21645b3453eec0b51031b67f23b45ec4acebb44e4781f473c661fb6e6R509) a lot more. When running this test locally with `fork_worker` enabled a phased restart is triggered ~10 times, which seems a bit excessive/unrealistic for 3k requests. `fork_worker` then adds to that overhead cause worker 0 doesn't accept connections until the refork cycle is complete.

Reply:
I see you created https://github.com/puma/puma/pull/3589 to address the above 👍